pkg/operator/controller/dns_status.go

```diff
@@ -70,14 +70,14 @@ func computeDNSDegradedCondition(oldCondition *operatorv1.OperatorCondition, clu
 		degradedCondition.Status = operatorv1.ConditionTrue
 		degradedCondition.Reason = "NoClusterIP"
 		degradedCondition.Message = "No ClusterIP assigned to DNS Service"
-	case ds.Status.NumberAvailable == 0:
+	case ds.Status.DesiredNumberScheduled == 0:
 		degradedCondition.Status = operatorv1.ConditionTrue
-		degradedCondition.Reason = "DaemonSetUnavailable"
-		degradedCondition.Message = "DaemonSet pod not running on any Nodes"
-	case ds.Status.NumberAvailable != ds.Status.DesiredNumberScheduled:
+		degradedCondition.Reason = "NoPodsScheduled"
+		degradedCondition.Message = "No CoreDNS pods are desired to be scheduled"
+	case ds.Status.DesiredNumberScheduled > 0 && ds.Status.NumberAvailable <= 1:
```

Comment:
@Miciah suggested `ds.Status.DesiredNumberScheduled - ds.Status.NumberAvailable > ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable` for this case, and I like that better than `NumberAvailable <= 1`, because it means that Kubernetes is violating our DaemonSet rollout condition, and presumably you could grow a cluster to be large enough that a single CoreDNS could not serve the whole cluster.